lib/org_date_rangeable.rb

```diff
@@ -0,0 +1,27 @@
+module OrgDateRangeable
+  def monthly_range(org:, start_date: Date.today.end_of_month, end_date: Date.today.end_of_month)
+    raise ArgumentError.new('missing org') unless org.present?
+    raise ArgumentError.new('missing start_date') unless start_date.respond_to?(:end_of_month)
+    raise ArgumentError.new('missing end_date') unless end_date.respond_to?(:end_of_month)
+
+    where(["org_id = ? and date >= ? and date <= ?", org.id, start_date, end_date]) 
+  end
+
+  class << self
+    def split_months_from_creation(org, &block)
+      starts_at = org.created_at
+      ends_at = starts_at.end_of_month
+      callable = block.nil? ? lambda{ |start_date, end_date| } : lambda{ | start_date, end_date| block.call(start_date, end_date) }
```

Comment:
If you use `Proc.new {}` here, you won't need to define the unused variables in the first block. 
Procs don't complain if parameters are missing.